src/app/clusters/time-synchronization-server/time-synchronization-server.h

```diff
@@ -0,0 +1,122 @@
+/**
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ * @file API declarations for time sync cluster.
+ */
+
+#pragma once
+
+#include "TimeSyncDataProvider.h"
+
+#include <app/server/Server.h>
+#include <app/util/af-types.h>
+#include <app/util/config.h>
+#include <lib/core/TLV.h>
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <lib/support/Span.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace TimeSynchronization {
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::TimeSynchronization;
+using namespace chip::app::Clusters::TimeSynchronization::Attributes;
+using chip::TimeSyncDataProvider;
+using chip::Protocols::InteractionModel::Status;
+
+struct timeZoneName
+{
+    char name[64];
+};
+enum class DSTState : uint8_t
+{
+    kInvalid = 0,
+    kActive  = 1,
+    kChanged = 2,
+    kStopped = 2,
+};
+
+class TimeSynchronizationServer
+{
+public:
+    void Init();
+
+    static TimeSynchronizationServer & Instance(void);
+    TimeSyncDataProvider GetDataProvider(void) { return mTimeSyncDataProvider; }
+
+    CHIP_ERROR SetTrustedTimeSource(DataModel::Nullable<TimeSynchronization::Structs::TrustedTimeSourceStruct::Type> tts);
+    CHIP_ERROR SetDefaultNTP(DataModel::Nullable<chip::CharSpan> & dntp);
+    /**
+     * @brief Sets TimeZone Attribute. Assumes the size of the list is already validated.
+     *
+     * @param tz TimeZone list
+     * @return CHIP_ERROR
```

Comment:
I've removed these 'documentations' completely. I also don't find them accurate and useful. Since the implementation has been changing a lot it is out of sync with the documentation. If needed, I will add documentation once the implementation settles.